scipy/linalg/basic.py

```diff
@@ -818,40 +840,111 @@ def lstsq(a, b, cond=None, overwrite_a=False, overwrite_b=False,
+            cond = np.finfo(lapack_func.dtype).eps * 100
+
+        if lapack_driver == 'gelss':
+            # Request of sizes
+            work,info = lapack_func_lwork_query(m,n,nrhs,cond)
+            lwork = np.int(np.real(work))
+            if info != 0:
+                raise ValueError("internal gelss driver lwork query "
+                                 "error, info is %i " % info)
+
+            v, x, s, rank, work, info = lapack_func(a1, b1, cond, lwork,
+                                                    overwrite_a=overwrite_a,
+                                                    overwrite_b=overwrite_b)
+
+        elif lapack_driver == 'gelsd':
+            if real_data:
+                # Request of sizes
+                work,iwork,info = lapack_func_lwork_query(m, n, nrhs, cond)
+                lwork = np.int(np.real(work))
+                iwork_size = iwork
+                if iwork_size == 0:
+                    # this is LAPACK bug 0038: dgelsd does not provide the
+                    # size of the iwork array in query mode.  This bug was
+                    # fixed in LAPACK 3.2.2, released July 21, 2010.
+                    raise LinAlgError("internal gelsd driver lwork query error,"
+                                     "required iwork dimension not returned.")
+                if info != 0:
+                    raise ValueError("internal gelsd driver lwork query error"
+                                     ", info is %i " % info)
+
+                x, s, rank, info = lapack_func(a1, b1, lwork,
+                                            iwork_size, cond, False, False)
+            else:  # complex data
+                # Request of sizes
+                work,rwork,iwork,info = lapack_func_lwork_query(m, n, nrhs, cond)
+                lwork = np.int(np.real(work))
+                rwork_size = np.int(rwork)
+                iwork_size = iwork
+                if info != 0:
+                    raise ValueError("internal gelsd driver lwork query error"
+                                     ", info is %i " % info)
+
+                x, s, rank, info = lapack_func(a1, b1,
+                            lwork,rwork_size, iwork_size, cond, False, False)
+        if info > 0:
+            raise LinAlgError("SVD did not converge in Linear Least Squares")
+        if info < 0:
+            raise ValueError('illegal value in %d-th argument of internal %s'
+                                                    % (-info, lapack_driver))
+        resids = np.asarray([], dtype=x.dtype)
+        if m > n:
+            x1 = x[:n]
+            if rank == n:
+                resids = np.sum(np.abs(x[n:])**2, axis=0)
+            x = x1
+        return x, resids, rank, s
+
+    elif lapack_driver == 'gelsy':
+        if cond is None:
+            cond = np.finfo(lapack_func.dtype).eps * 100
+
+        # Request of sizes
+        work,info = lapack_func_lwork_query(m, n, nrhs, cond)
+        lwork = np.int(np.real(work))
```

Comment:
This is probably just my ignorance: why is this being cast to real and then numpy int?